include/bluetooth/bluetooth.h

```diff
@@ -772,9 +803,218 @@ int bt_le_ext_adv_get_info(const struct bt_le_ext_adv *adv,
  *  @param adv_type Type of advertising response from advertiser.
  *  @param buf Buffer containing advertiser data.
  */
+
 typedef void bt_le_scan_cb_t(const bt_addr_le_t *addr, int8_t rssi,
 			     uint8_t adv_type, struct net_buf_simple *buf);
 
+/**
+ * @brief Set or update the periodic advertisement parameters.
+ *
+ * The periodic advertisement parameters can only be set or updated on an
+ * extended advertisement set which is neither scannable, connectable or
+ * anonymous.
+ *
+ * @param adv	Advertising set object.
+ * @param param	Advertising parameters.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_set_param(struct bt_le_ext_adv *adv,
+			    const struct bt_le_per_adv_param *param);
+
+/**
+ * @brief Set or update the periodic advertisement data.
+ *
+ * The periodic advertisement data can only be set or updated on an
+ * extended advertisement set which is neither scannable, connectable or
+ * anonymous.
+ *
+ * @param adv		Advertising set object.
+ * @param data		Advertising data.
+ * @param data_len	Advertising data length.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_set_data(const struct bt_le_ext_adv *adv, const uint8_t *data,
+			   uint8_t data_len);
+
+/**
+ * @brief Enables or disables periodic advertisement.
+ *
+ * Enabling the periodic advertisement can be done independently of extended
+ * advertisement, but both periodic advertisement and extended advertisement
+ * shall be enabled before any data is sent.
+ * Enabling the periodic advertisement while it is already
+ * enabled updates the random address.
+ *
+ * @param adv		Advertising set object.
+ * @param enable	Whether to enable or disable periodic advertisement.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_enable(struct bt_le_ext_adv *adv, bool enable);
+
+struct bt_le_per_adv_sync_synced_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** Periodic advertisement interval (N * 1.25MS) */
+	uint16_t interval;
+
+	/** Periodic advertisement advertiser clock accuracy (ppm) */
+	uint16_t clock_accuracy;
+
+	/** Advertiser PHY */
+	uint8_t phy;
+};
+
+struct bt_le_per_adv_sync_term_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+};
+
+struct bt_le_per_adv_sync_recv_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** The TX power of the advertisement. */
+	int8_t tx_power;
+
+	/** The RSSI of the advertisement excluding any CTE. */
+	int8_t rssi;
+
+	/** The Constant Tone Extension (CTE) of the advertisement */
+	uint8_t cte;
```

Comment:
cte_type?

I think we should create some definitions for the defined valuse 0x00 - 0x02

Reply:
Agreed. I put them as an enum next to this struct.